dask_ml/model_selection/_search.py

```diff
@@ -286,16 +317,98 @@ def normalize_params(params):
 
     return fields, tokens, params2
 
+def _generate_fit_params_key_vals(fit_params, keys_filtered=None):
+    '''
+        _generate_fit_params_key_vals returns keys and values in (name,full_name)
+        and values format expected in CVcache functions
 
-def _get_fit_params(cv, fit_params, n_splits):
-    if not fit_params:
-        return [(n, None) for n in range(n_splits)]
+        fit_params: dict[str,Any] fit params dictionary from input to fit() call
+        keys_filtered: list[str] keys to filter from fit_params from output
+
+        returns: ( (fit_parameter_name,fit_parameter_full_name), fit_param_value )
+    '''
     keys = []
     vals = []
     for name, (full_name, val) in fit_params.items():
+        if keys_filtered is not None and name not in keys_filtered:
+            continue
         vals.append(val)
         keys.append((name, full_name))
-    return [(n, (cv_extract_params, cv, keys, vals, n)) for n in range(n_splits)]
+    return keys, vals
+
+def _check_fit_params_key_used(fp,key):
+    '''
+        _check_fit_params_key_used checks a fit_parameter is actually available
+        and not None.
+
+        fp: Dict[str,Any] fit_params
+        key: str fit parameter name
+
+        return bool status of fit parameter
+    '''
+    if key in fp:
```

Comment:
logic is a little bit complicated here, can you simply this function? what cases should return True and rest of cases should return False.

Reply:
added more comments in latest commit